plugins/power/tasmota_

```diff
@@ -0,0 +1,155 @@
+
+if [ "$1" = "autoconf" ]; then
+    echo yes && exit 0
+fi
+
+if [ "$1" = "suggest" ]; then
+    nodeattr -n 'tasmota&&powermon' | while read device ; do
+	for i in voltage power powerfactor current energy; do
+	    echo "${device}_${i}"
+	done
+    done
+    exit
+fi
+
+voltage() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Voltage: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_vlabel Volts"
+        echo "graph_args --base 1000 -l 0"
+
+	echo "Volts.label Volts"
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        volts=$( echo "$res" | jq -c '.StatusSNS.ENERGY.Voltage' )
+        echo Volts.value $volts
+    fi
+}
+
+current() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Current: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_vlabel Amps"
+        echo "graph_args --base 1000 -l 0"
+
+	echo "Current.label Current"
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        current=$( echo "$res" | jq -c '.StatusSNS.ENERGY.Current' )
+        echo Current.value $current
+    fi
+}
+
+power() {
+    if [ "$1" = "config" ]; then
+
+	echo "graph_title Tasmota Power: $SWITCH"
+        echo "graph_category sensors"
+	echo "graph_vlabel Watts"
+        echo "graph_args --base 1000 -l 0"
+
+	for i in Power ApparentPower ReactivePower ; do
+	    echo "${i}.label $i"
+	    echo "${i}.type GAUGE"
+	    echo "${i}.min 0"
+	done
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+	for i in Power ApparentPower ReactivePower ; do
+            power=$( echo "$res" | jq -c '.StatusSNS.ENERGY.'$i )
+            echo $i.value $power
+        done
+    fi
+}
+
+powerfactor() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Power factor: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_vlabel Power Factor"
+        echo "graph_args --base 1000 -l 0"
+
+	echo "PowerFactor.min 0"
+	echo "PowerFactor.max 1"
+	echo "PowerFactor.label Power Factor"
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        powerfactor=$( echo "$res" | sed 's/.*"Factor":\([0-9.]*\)[^0-9.]*.*/\1/' )
+        echo PowerFactor.value $powerfactor
+    fi
+}
+
+energy() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Energy: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_args --base 1000"
+
+        echo "graph_vlabel kWh"
+	echo "Energy.label Energy"
+        echo "Energy.draw AREA"
+
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        energy=$( echo "$res" | sed 's/.*"Total":\([0-9.]*\)[^0-9.]*.*/\1/' )
+        echo Energy.value $energy
+    fi
+}
+
+
+[ "$1" = "config" ] && echo "graph_category sensors"
+
+case "$FUNCTION" in
```

Comment:
Please emit an error message on `stdout` end exit with a failure, if an unknown function is requested.

Reply:
Done